main.cpp

```diff
@@ -41,22 +41,458 @@ If you need inspiration for what to write, take a look at previously approved st
 /*
  UDT 1:
  */
+#include <iostream>
+#include <string>
+#include <cmath>
+
+using namespace std;
+
+struct Synth
+{
+    struct UIControls
+    {
+        UIControls(int Knob_, int Button_) : Knob(Knob_), Button(Button_)
+        {
+            cout << "construct UIControls" << endl;
+        }
+        ~UIControls()
+        {
+            cout << "destruct UIControls" << endl; 
+        }
+        float setKnob(float newknobPosition);
+        bool buttonPress(bool newButtonState);
+        int updateLEDIndicator();
+
+        int Knob;
+        float knobPosition {0.f};
+        int Button;
+        bool buttonState = false;
+        int ledGainIndicator = 0;
+        int buttonCounter = 0;
+    };
+
+    UIControls uiControls{3, 4};
+
+    Synth();
+    ~Synth();
+
+    int modulateBeatFrequency( int newFreq1, int newFreq2 );
+    float changeAttack(float newAttack);
+    float changeRelease(float newRelease);
+    void setGain(int newLevel);
+
+    int frequency1;
+    int frequency2;
+    int frequencyOutput;
+    int gain;
+    float attack;
+    float release;
```

Comment:
initialize your members in-class whenever you can.
it makes your code simpler
- [ ] revise